lib/case.js

```diff
@@ -2,59 +2,66 @@
 
 const common = require('metarhia-common');
 
-module.exports = (metatests) => {
-
-  metatests.case = (
-    caption, // Case caption
-    list // hash of Array
-    // Hint: hash keys are function and method names
-    // Array contains call parameters
-    // last Array item is an expected result (to compare)
-    // or function (pass result to compare)
-  ) => {
-    metatests.results.modules++;
-
-    let conditions, target, targetType, targetValue, definition, msg, cmp,
-        parameters, sParameters, expected, sExpected, expectedType,
-        result, sResult, functionPath, className, methodName, functionName;
-
-    const classes = [];
-
-    for (functionName in list) {
-      conditions = list[functionName];
+const { Test } = require('./test');
+const { globalNamespace } = require('./namespaces');
+const reporters = require('./report');
+
+class DeclarativeTest extends Test {
+  constructor(caption, list, namespaces) {
+    super(caption, () => this._runCase(this.list));
```

Comment:
I don't line naming with `_`. Why do we need `_runCase` and `checkCase` ?

Reply:
Well, `_runCase` I will fix that's a leftover from the previous thought-train.
`checkCase` is just a `case`, but because `case` is a keyword I cannot name the function as such. As for the `case` function itself, it's needed to avoid having user manually create and configure `DeclarativeTest`. This greatly improves user experience if he doesn't need some specific customization.